scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/definition/component/ComponentsFromProvidersExtractor.scala

```diff
@@ -26,6 +27,107 @@ object ComponentsFromProvidersExtractor {
 
 }
 
+final case class Components(
+    components: List[ComponentDefinitionWithImplementation],
+    // components without enrichments from an additional provider
+    basicComponents: List[ComponentDefinitionWithImplementation]
+) {
+
+  def withComponents(componentsToAdd: Components): Components = {
+    Components.combine(this, componentsToAdd)
+  }
+
+  def filter(predicate: ComponentDefinitionWithImplementation => Boolean): Components = {
+    copy(
+      components = components.filter(predicate),
+      basicComponents = components.filter(predicate),
+    )
+  }
+
+}
+
+object Components {
+  val empty: Components = Components(List.empty, List.empty)
+
+  sealed trait ComponentDefinitionExtractionMode
+
+  object ComponentDefinitionExtractionMode {
+    case object FinalDefinition          extends ComponentDefinitionExtractionMode
+    case object FinalAndBasicDefinitions extends ComponentDefinitionExtractionMode
+  }
+
+  def withComponent(
+      componentName: String,
+      component: Component,
+      configFromDefinition: ComponentConfig,
+      componentsUiConfig: ComponentsUiConfig,
+      determineDesignerWideId: ComponentId => DesignerWideComponentId,
+      additionalConfigsFromProvider: Map[DesignerWideComponentId, ComponentAdditionalConfig],
+      componentDefinitionExtractionMode: ComponentDefinitionExtractionMode
+  ): Components = {
+
+    def extractComponentDefinition(configFromProvider: Map[DesignerWideComponentId, ComponentAdditionalConfig]) = {
+      ComponentDefinitionExtractor
+        .extract(
+          componentName,
+          component,
+          configFromDefinition,
+          componentsUiConfig,
```

Comment:
I thought that we agreed that for `basicComponents` we also skip `componentsUiConfig`. It is easier to think about component definitions if both `componentsUiConfig` and `configFromProvider` are treated the same.

Reply:
Ok, but now I think that we should also rename `enrichedWithAdditionalConfig` to `enrichedWithUiConfig` and change comments such as `components without enrichments from an additional provider`.